agent.py

```diff
@@ -895,7 +893,12 @@ def findNeighborhood(self, newCell=None):
                 neighborhood.append(neighbor)
         neighborhood.append(self)
         if newCell == None:
+            # Shuffle cells for movement considerations
+            random.shuffle(allCells)
             self.neighborhood = neighborhood
+        # This neighborhood should not be used for movement,
+        # only for calculating len(agent.findNeighborhood(cell))
+        # in *HalfLookahead decision models.
```

Comment:
These comments can be removed since they're explaining design-level rationale as opposed to immediate code comprehension. It should be noted your comments are factually correct. You found a good performance optimization.